### PR TITLE
Adding principal regarding querying/returning predicate inverses.  

### PR DESCRIPTION
**Pull request by sierra-moxon**
- PR reopened to contend with the readme vs. README issue.
- refactored 'should' to 'must'.
- added qualification that this principal applies to KGX files and TrAPI messages as well.

### CODE REVIEW

**Comment by codewarrior2000** [comment]
Perhaps it might be better to change in the README the wording from "This principal also applies ..." to "This principle also applies ..."

File: README.md [hunk 1]
```diff
@@ -69,7 +68,7 @@ This README documents the current strawman architecture.  Changes must be made v
     1. The SmartAPI registry will allow components to find all KPs by querying for biolink predicates. The SmartAPI registry will allow components to query by predicate from biolink-understood vocabularies, and return KPs that provide such metadata.
 1. A continuous integration framework will consume metadata from the registry, and provide automated testing and reports.
 1. Both KPs and ARAs should acquire and transmit provenance information to the fullest possible extent.
-
+1. When querying and returning results with predicates, KPs and ARAs must be queried using the 'canonical' predicate (as opposed to its inverse), and must return the 'canonical' predicate.  There will be two ways to identify the 'canonical' predicate in the biolink-model: canonical translator predicates will not be tagged with the 'inverse:' attribute, and canonical predicates will be tagged with an "annotations" flag with the tag: "biolink:canonical_predicate" and value: "True".  This principle also applies to KGX files and TrAPI messages.
```
Comment (patrickkwang) [inline README.md:71]:
```suggestion
1. When querying and returning results with predicates, KPs and ARAs must be queried using the 'canonical' predicate (as opposed to its inverse), and must return the 'canonical' predicate.  There will be two ways to identify the 'canonical' predicate in the biolink-model: canonical translator predicates will not be tagged with the 'inverse:' attribute, and canonical predicates will be tagged with an "annotations" flag with the tag: "biolink:canonical_predicate" and value: "True".  This principle also applies to KGX files and TRAPI messages.
```
The R stands for "reasoner".
🤷

**Review by patrickkwang — CHANGES_REQUESTED** [review]

**Review by vdancik — APPROVED** [review]

**Review by bill-baumgartner — APPROVED** [review]

**Review by edeutsch — APPROVED** [review]

**Review by karthiksoman — APPROVED** [review]

**Review by suihuang-ISB — APPROVED** [review]

**Review by newgene — APPROVED** [review]

**Review by rtroper — APPROVED** [review]

**Review by karthiksoman — APPROVED** [review]

**Review by bill-baumgartner — APPROVED** [review]

**Review by vdancik — APPROVED** [review]

**Review by rtroper — APPROVED** [review]

**Review by patrickkwang — APPROVED** [review]

**Comment by CaseyTa** [comment]
@sierra-moxon, could you please clarify where it says 

> "canonical translator predicates will not be tagged with the 'inverse:' attribute"

Does this mean that only the non-canonical predicate will specify which predicate is its inverse? 

So take `biolink:superclass_of` and `biolink:subclass_of`. Currently, both have their inverses defined as each other. If `biolink:subclass_of` is defined as the canonical direction, then would it no longer specify `inverse: superclass of`, while `biolink:superclass_of` would keep its definition of `inverse: subclass of`? If this is the case, it might be slightly more inconvenient to find the inverse of a canonical predicate. It also seems unnecessary to remove the inverse if the canonical predicate will also be specified by `biolink:canonical_predicate`. 

This is just a very minor quibble, and either way, I'll approve of the changes.

**Review by CaseyTa — APPROVED** [review]

**Comment by sierra-moxon** [comment]
> @sierra-moxon, could you please clarify where it says
> 
> > "canonical translator predicates will not be tagged with the 'inverse:' attribute"
> 
> Does this mean that only the non-canonical predicate will specify which predicate is its inverse?
> 
> So take `biolink:superclass_of` and `biolink:subclass_of`. Currently, both have their inverses defined as each other. If `biolink:subclass_of` is defined as the canonical direction, then would it no longer specify `inverse: superclass of`, while `biolink:superclass_of` would keep its definition of `inverse: subclass of`? If this is the case, it might be slightly more inconvenient to find the inverse of a canonical predicate. It also seems unnecessary to remove the inverse if the canonical predicate will also be specified by `biolink:canonical_predicate`.
> 
> This is just a very minor quibble, and either way, I'll approve of the changes.

Hi Casey!  Yep, your example is inline with the intention of this principle.  @cmungall's original idea for declaring the inverse was to only curate the predicate in one direction (ie: only the non-canonical predicate would have the 'inverse' attribute populated).  We added the biolink:canonical_predicate to make this a bit simpler to pull out from the canonical predicate itself.  We did open a ticket in the BMT repo, to make a helper function for this as well: https://github.com/biolink/biolink-model-toolkit/issues/30.

**Review by cbizon — APPROVED** [review]

**Pull request merged**